src/cpp/rtps/builtin/data/ParticipantProxyData.cpp

```diff
@@ -446,16 +446,15 @@ bool ParticipantProxyData::writeToCDRMessage(
 bool ParticipantProxyData::readFromCDRMessage(
         CDRMessage_t* msg,
         bool use_encapsulation,
-        const NetworkFactory& network,
-        bool is_shm_transport_available,
+        NetworkFactory& network,
         bool should_filter_locators,
         fastdds::rtps::VendorId_t source_vendor_id)
 {
     auto param_process =
-            [this, &network, &is_shm_transport_available, &should_filter_locators, source_vendor_id](
+            [this, &network, &should_filter_locators, source_vendor_id](
         CDRMessage_t* msg, const ParameterId_t& pid, uint16_t plength)
             {
-                static_cast<void>(source_vendor_id);
+                m_VendorId = source_vendor_id;
```

Comment:
Double check this change would not introduce unwanted implications. Related to internal task #21423

Reply:
I think it makes sense, we set the `m_VendorId` to whatever is in the `change->vendor_id` and then, if the `PID_VENDORID` is specified, we set it to that value.

Reply:
I was not sure this was super correct since I though the method is supposed to read the content of a data p/w/r , setting its attributes given the fields parsed. Anyways, I did not find any behavior change given the current logic but I wanted someone else to verify just in case. And this will probably be addressed in the future when tackling the related task I made reference to.